h/views/activity.py

```diff
@@ -90,27 +92,25 @@ def user_link(userid):
         }
 
 
-@view_defaults(route_name='activity.group_search',
-               renderer='h:templates/activity/search.html.jinja2')
+@view_defaults(route_name='group_read',
+               renderer='h:templates/activity/search.html.jinja2',
+               custom_predicates=[custom_predicates.has_read_permission,
+                                  custom_predicates.feature('search_page')])
 class GroupSearchController(SearchController):
-    """View callables unique to the "activity.group_search" route."""
+    """View callables unique to the "group_read" route."""
+
+    def __init__(self, group, request):
+        super(GroupSearchController, self).__init__(request)
+        self.group = group
+        check_slug(self.group, self.request)
 
     @view_config(request_method='GET')
     def search(self):
         result = super(GroupSearchController, self).search()
 
-        pubid = self.request.matchdict['pubid']
-        result['opts'] = {'search_groupname': pubid}
-
-        try:
-            group = (self.request.db.query(models.Group)
-                     .filter_by(pubid=pubid).one())
-        except exc.NoResultFound:
-            return result
```

Comment:
Now that we're using traversal the view no longer needs to look up the group in the db (it's available as `self.group`) or handle the case where the group doesn't exist (Pyramid won't call the view in this case)